app/controllers/spree/api/retailops/settlement_controller.rb

```diff
@@ -62,14 +62,30 @@ def mark_complete
           render text: @settlement_results.to_json
         end
 
+        # This gets called by RetailOps after a returned item has been
+        # received and accepted at the warehouse. We need to look elsehwere
+        # for hook when returned item has been received but can not be restocked.
+        #
+        # We first look for a `RetailopsReimbursementHandler` class to see if the host
+        # application has defined any custom logic. If they have, we simply
+        # pass the parameters to the defined `Reimbursement.from_retailops_add_refund`
+        # factory method and call `RetailopsReimbursementHandler#create_reimbursement`.
         def add_refund
-          ActiveRecord::Base.transaction do
-            find_order
-            assert_return params
-            @order.update!
+          if defined?(RetailopsReimbursementHandler)
+            reimbursement = RetailopsReimbursementHandler.from_retailops_add_refund(
+              params['settlement']
```

Comment:
Actually I am referring to `params['settlement']`.  Here's an `add_refund` call we recently sent for Huckberry:

```json
{
    "shipping_amt" : 0,
    "subtotal_amt" : 179.98,
    "product_amt" : 179.98,
    "options" : {},
    "refund_amt" : "195.73",
    "tax_amt" : 15.75,
    "order_refnum" : "R270263973",
    "rma_id" : "20924",
    "return_items" : [
        {
            "order_item_id" : "560864",
            "channel_refnum" : "859700",
            "sku" : "124465",
            "quantity" : "1"
        }
    ],
    "refund_items" : [
        {
            "refnum" : "859700",
            "amount" : "195.73",
            "quantity" : "1",
            "label" : "Credit (return_item 24053)",
            "id" : "859700"
        }
    ],
    "return_id" : "19925"
}
```

There's no `settlement` there and I'm not sure what, if anything, could be adding it before your code runs.